src/main/java/com/github/manolo8/darkbot/extensions/plugins/PluginHandler.java

```diff
@@ -260,5 +266,22 @@ private void testSignature(Plugin plugin, JarFile jar) throws IOException {
         }
     }
 
+    private String[] fetchFeatureList(Plugin plugin, JarFile jar) {
+        URLClassLoader loader = new URLClassLoader(new URL[]{ plugin.getJar() });
+        List<String> features = new ArrayList<>();
+        Enumeration<JarEntry> entries = jar.entries();
+        while (entries.hasMoreElements())
+            try {
+                JarEntry jarEntry = entries.nextElement();
+                if (jarEntry.getName().endsWith(".class")) {
+                    String className = jarEntry.getName().replace("/", ".").replace(".class", "");
+                    if(loader.loadClass(className).isAnnotationPresent(Feature.class))
+                        features.add(className);
+                }
```

Comment:
This is essentially going to load **every single class**, in the jar, regardless of if it's needed or not, just to search for features, we have a basePackage in the plugin.json (the plugin definition) and we should use it. Search/load only classes which are inside that base package, avoid iterating or loading all dependencies a plugin includes, they're not gonna be features